sdk/resources/mgmt/src/main/java/com/azure/management/resources/fluentcore/arm/collection/implementation/CreatableResourcesImpl.java

```diff
@@ -79,12 +79,12 @@ protected CreatableResourcesImpl() {
      *
      * @param <ResourceT> the type of the resources in the batch.
      */
-    private class CreatedResourcesImpl<ResourceT extends Indexable>
+    private static class CreatedResourcesImpl<ResourceT extends Indexable>
             extends HashMap<String, ResourceT>
             implements CreatedResources<ResourceT> {
-        private final ClientLogger logger = new ClientLogger(this.getClass());
+        private final transient ClientLogger logger = new ClientLogger(this.getClass());
```

Comment:
Please detail on what happens here. `transient` should not be used lightly.

Reply:
The spotbugs said the CreatableResourcesImpl has two variable that cannot be serialized. After my search, it seems add transient can fix it.

Reply:
Could you see if it is possible to convert the class to composition (use a HashMap var, instead of extends HashMap, then delegate other Map method to that var)? This way we should be able to get rid of `Serializable` interface.

Reply:
Changed extends to private var. Done